connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/arrow/ArrowSerializer.scala

```diff
@@ -0,0 +1,529 @@
+      case (DateEncoder(false), v: DateDayVector) =>
+        new FieldSerializer[java.sql.Date, DateDayVector](v) {
+          override def set(index: Int, value: java.sql.Date): Unit =
+            vector.setSafe(index, DateTimeUtils.fromJavaDate(value))
+        }
+      case (LocalDateEncoder(false), v: DateDayVector) =>
+        new FieldSerializer[LocalDate, DateDayVector](v) {
+          override def set(index: Int, value: LocalDate): Unit =
+            vector.setSafe(index, DateTimeUtils.localDateToDays(value))
+        }
+      case (TimestampEncoder(true) | InstantEncoder(true), v: TimeStampMicroTZVector) =>
+        new FieldSerializer[Any, TimeStampMicroTZVector](v) {
+          override def set(index: Int, value: Any): Unit =
+            vector.setSafe(index, DateTimeUtils.anyToMicros(value))
+        }
+      case (TimestampEncoder(false), v: TimeStampMicroTZVector) =>
+        new FieldSerializer[java.sql.Timestamp, TimeStampMicroTZVector](v) {
+          override def set(index: Int, value: java.sql.Timestamp): Unit =
+            vector.setSafe(index, DateTimeUtils.fromJavaTimestamp(value))
+        }
+      case (InstantEncoder(false), v: TimeStampMicroTZVector) =>
+        new FieldSerializer[Instant, TimeStampMicroTZVector](v) {
+          override def set(index: Int, value: Instant): Unit =
+            vector.setSafe(index, DateTimeUtils.instantToMicros(value))
+        }
+      case (LocalDateTimeEncoder, v: TimeStampMicroVector) =>
+        new FieldSerializer[LocalDateTime, TimeStampMicroVector](v) {
+          override def set(index: Int, value: LocalDateTime): Unit =
+            vector.setSafe(index, DateTimeUtils.localDateTimeToMicros(value))
+        }
+
+      case (OptionEncoder(value), v) =>
+        new Serializer {
+          private[this] val delegate: Serializer = serializerFor(value, v)
+          override def write(index: Int, value: Any): Unit = value match {
+            case Some(value) => delegate.write(index, value)
+            case _ => delegate.write(index, null)
+          }
+        }
+
+      case (ArrayEncoder(element, _), v: ListVector) =>
+        val elementSerializer = serializerFor(element, v.getDataVector)
+        val toIterator = { array: Any =>
+          mutable.WrappedArray.make(array.asInstanceOf[AnyRef]).iterator
+        }
+        new ArraySerializer(v, toIterator, elementSerializer)
+
+      case (IterableEncoder(tag, element, _, lenient), v: ListVector) =>
+        val elementSerializer = serializerFor(element, v.getDataVector)
+        val toIterator: Any => Iterator[_] = if (lenient) {
+          {
+            case i: scala.collection.Iterable[_] => i.toIterator
+            case l: java.util.List[_] => l.iterator().asScala
+            case a: Array[_] => a.iterator
+            case o => unsupportedCollectionType(o.getClass)
+          }
+        } else if (isSubClass(Classes.ITERABLE, tag)) { v =>
+          v.asInstanceOf[scala.collection.Iterable[_]].toIterator
```

Comment:
`.toIterator` -> `iterator`, `.toIterator`  is deprecated after Scala 2.13

Reply:
Done.